content/gitlab-ci.md

```diff
@@ -0,0 +1,217 @@
+executing, the repository has a blue marker. This is replaced with a green
+check mark, once the test succeeds:
+
+```{figure} img/gl_green_check_mark.png
+:alt: Verify that the test passed
+
+Green check means passed.
+```
+
+Also browse the "Pipelines" tab and look at the steps there and their output.
+
+
+### Step 5: Add a test which reveals a problem
+
+After you committed the workflow file, your Gitlab repository will be ahead of
+your local cloned repository.  Update your local cloned repository:
+
+```bash
+$ git pull origin main
+```
+
+Next uncomment the code in `example.py` under "step 5", commit, and push.
+Verify that the test suite now fails on the "Actions" tab.
+
+
+### Step 6: Open an issue on Gitlab
+
+Open a new issue in your repository on the broken test on Gitlab.
+The plan is that your colleague will fix the issue.
+
+
+### Step 7: Fork and clone the repository of your colleague
+
+Fork the repository using the Gitlab web interface. Make sure you clone the
+fork after you have forked it. Do not clone your colleague's repository
+directly.
+
+```bash
+$ git clone https://gitlab.com/your-username/the-repository.git
+```
+
+
+### Step 8: Fix the broken test
+
+After you have fixed the code,
+commit the following commit message `"restore function subtract; fixes #1"` (assuming that you try to fix issue number 1).
+
+Then push to your fork.
+
+
+### Step 9: Open a merge request
+
+Then before accepting the merge request from your colleague, observe
+how Gitlab-CI automatically tested the code. Add a description to the merge request with a 
+"Closes #NUMBEROFYOURISSUE".
+
+
+### Step 10: Accept the merge request
+
+Observe how accepting the merge request automatically closes the issue (provided
+the merge request message contained the correct issue number).
+
+See also
+[closing issues using keywords](https://docs.gitlab.com/ee/user/project/issues/managing_issues.html#closing-issues).
+
+Discuss whether this is a useful feature. And if it is, why do you think is it useful?
+
+
+### Discussion
+
+Finally, we discuss together about our experiences with this exercise.
+
+
+```{challenge} Optional exercise
+In the [Social coding and open software](https://cicero.xyz/v3/remark/0.14.0/github.com/coderefinery/social-coding/master/talk.md/)
+lesson we learn how important it is to add a LICENSE file.
+
+Your goal:
+- You discover that your coworker's repository does not have a LICENSE file.
+- Open an issue and suggest a LICENSE.
+- Then add a LICENSE via a pull/merge request, referencing the issue number.
+```
+
+---
+
+## Where to go from here
+
+- This example was using Python but you can achieve the same automation for R or Fortran or C/C++ or other languages
+- GitHub Actions has a [Marketpace](https://github.com/marketplace?type=actions) which offer wide range of automatic workflows
+- Check out the [tutorial](https://coderefinery.github.io/testing/gh-actions/) about GitHub Actions
```

Comment:
is there a corresponding page for GitLab CI? Maybe just link to the docs: https://docs.gitlab.com/runner/

Reply:
Yeah not really it has some default template workflows, but they are a lot more complex, using caching etc.. So I am a bit hesitant to link them here.